ta4j-core/src/main/java/org/ta4j/core/BaseTradingRecord.java

```diff
@@ -203,4 +238,18 @@ private void recordOrder(Order order, boolean isEntry) {
             currentTrade = new Trade(startingType);
         }
     }
+
+    /**
+     * @return the starting index of the trading session
+     */
+    public int getStartIndex() {
+		return startIndex;
+	}
+
+    /**
+     * @return the finishing index of the trading session
+     */
+	public int getFinishIndex() {
+		return finishIndex;
+	}
```

Comment:
`getStartIndex()` and ` getFinishIndex()` return a null pointer if this `BaseTradingRecord` was not initialized by your new constructor, but with one of the old once that do not initialize the fields.
- You could initialize them with `Integer.MAX_VALUE`
- or check in getter` if startIndex == null return Integer.MAX_VALUE`

Reply:
I would prefer to initialize them with `finishIndex = Integer.MAX_VALUE` and `startIndex = 0`

Reply:
Now initialized in BaseTradingRecord#BaseTradingRecord(Order.OrderType), which is in the constructor chain for all other constructors.  When constructed with indices, those fields are set after being initialized as above.